4.1rc1: test_wtbarr_print fails during test

### Description

When I compile and and test 4.1rc1 for Debian (i.e. with an external wcslib; version 7.2), I get a failure during the build time test.

### Expected behavior

All tests pass

### Actual behavior

The following failure appears:

```
______________________________ test_wtbarr_print _______________________________

tab_wcs_2di = WCS Keywords

Number of WCS axes: 2
CTYPE : 'RA---TAB'  'DEC--TAB'  
CRVAL : 1.0  1.0  
CRPIX : 1.0  1.0  
PC1_1 PC1_2  : 1.0  0.0  
PC2_1 PC2_2  : 0.0  1.0  
CDELT : 1.0  1.0  
NAXIS : 150  200
capfd = <_pytest.capture.CaptureFixture object at 0x7f86e08ef190>

    def test_wtbarr_print(tab_wcs_2di, capfd):
        tab_wcs_2di.wcs.wtb[0].print_contents()
        captured = capfd.readouterr()
        s = str(tab_wcs_2di.wcs.wtb[0])
        lines = s.split('\n')
>       assert captured.out == s
E       AssertionError: assert '\n\n\n\n\n\n...: 0x49e43b8\n' == '     i: 1\n  ...: 0x49e43b8\n'
E         - 
E         - 
E         - 
E         - 
E         - 
E         - 
E         - ...
E         
E         ...Full output truncated (39 lines hidden), use '-vv' to show

astropy/wcs/tests/test_wtbarr.py:51: AssertionError
= 1 failed, 14395 passed, 440 skipped, 53 xfailed, 12 xpassed in 307.52 seconds =
E: pybuild pybuild:352: test: plugin distutils failed with: exit code=1: cd '/build/astropy-4.1~rc1/.pybuild/cpython3_3.8/build'; python3.8 -m pytest 
dh_auto_test: error: pybuild --test --test-pytest -i python{version} -p 3.8 returned exit code 13
```


### Steps to Reproduce

Build and test astropy on Debian sid, with  `ASTROPY_USE_SYSTEM_ALL=1`.

### System Details
<!-- Even if you do not think this is necessary, it is useful information for the maintainers.
Please run the following snippet and paste the output below:
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("Numpy", numpy.__version__)
import astropy; print("astropy", astropy.__version__)
import scipy; print("Scipy", scipy.__version__)
import matplotlib; print("Matplotlib", matplotlib.__version__)
-->
```
Python 3.8.3 (default, May 14 2020, 11:03:12) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform; print(platform.platform())
Linux-5.4.0-4-amd64-x86_64-with-glibc2.29
>>> import sys; print("Python", sys.version)
Python 3.8.3 (default, May 14 2020, 11:03:12) 
[GCC 9.3.0]
>>> import numpy; print("Numpy", numpy.__version__)
Numpy 1.18.4
>>> import scipy; print("Scipy", scipy.__version__)
Scipy 1.4.1
>>> import matplotlib; print("Matplotlib", matplotlib.__version__)
```
wcslib is 7.2

--- a/astropy/wcs/tests/data/3d_cd.hdr
+++ b/astropy/wcs/tests/data/3d_cd.hdr
@@ -1,16 +1,1 @@
-CD1_2   =            -3.72E-05                                                  
-CD1_3   =                    0                                                  
-CD1_1   =            -4.12E-05                                                  
-CUNIT3  = 'nm      '                                                            
-CUNIT2  = 'deg     '                                                            
-CTYPE1  = 'RA---TAN'                                                            
-NAXIS   =                    3                                                  
-CTYPE3  = 'AWAV    '                                                            
-CD2_1   =            -3.72E-05                                                  
-CTYPE2  = 'DEC--TAN'                                                            
-CD2_3   =                    0                                                  
-CD2_2   =             4.12E-05                                                  
-CUNIT1  = 'deg     '                                                            
-CD3_1   =                    0                                                  
-CD3_2   =                    0                                                  
-CD3_3   =                  0.2                                                  
+CD1_2   =            -3.72E-05                                                  CD1_3   =                    0                                                  CD1_1   =            -4.12E-05                                                  CUNIT3  = 'nm      '                                                            CUNIT2  = 'deg     '                                                            CTYPE1  = 'RA---TAN'                                                            NAXIS   =                    3                                                  CTYPE3  = 'AWAV    '                                                            CD2_1   =            -3.72E-05                                                  CTYPE2  = 'DEC--TAN'                                                            CD2_3   =                    0                                                  CD2_2   =             4.12E-05                                                  CUNIT1  = 'deg     '                                                            CD3_1   =                    0                                                  CD3_2   =                    0                                                  CD3_3   =                  0.2                                                  

--- a/astropy/wcs/tests/test_wcs.py
+++ b/astropy/wcs/tests/test_wcs.py
@@ -791,11 +791,10 @@ def test_printwcs(capsys):
         captured = capsys.readouterr()
         assert 'WCS Keywords' in captured.out
     h = get_pkg_data_contents('data/3d_cd.hdr', encoding='binary')
-    with pytest.warns(AstropyUserWarning):
-        w = wcs.WCS(h)
-        w.printwcs()
-        captured = capsys.readouterr()
-        assert 'WCS Keywords' in captured.out
+    w = wcs.WCS(h)
+    w.printwcs()
+    captured = capsys.readouterr()
+    assert 'WCS Keywords' in captured.out
 
 
 def test_invalid_spherical():

